tests/test_logger.py

```diff
@@ -0,0 +1,149 @@
+
+from iquip.apps import logger
+
+class _SignallerTest(unittest.TestCase):
+    """Unit tests for the _Signaller class."""
+
+    def setUp(self):
+        self.qapp = QApplication([])
+
+    def tearDown(self):
+        del self.qapp
+
+    def test_signal_connection(self):
+        with mock.patch('iquip.apps.logger._Signaller.signal') as mocked_signal:
+            app = logger.LoggerApp(name="name", parent=QObject())
+            mocked_signal.connect.assert_called_once_with(app.addLog)
+
+
+class LoggingHandlerTest(unittest.TestCase):
+    """Unit tests for the LoggingHandler class."""
+
+    def setUp(self):
+        self.qapp = QApplication([])
+
+    def tearDown(self):
+        del self.qapp
+
+    def test_handler_connection(self):
+        with mock.patch('iquip.apps.logger._Signaller.signal') as mocked_signal:
+            app = logger.LoggerApp(name="name", parent=QObject())
+            mocked_signal.connect.assert_called_once_with(app.addLog)
+
+    def test_emit(self):
+        with mock.patch('iquip.apps.logger.LoggingHandler.emit') as mocked_method:
+            app = logger.LoggerApp(name="name", parent=QObject())
+            test_logger = logger.logger
+            app.setLevel("INFO")
+            test_logger.info("hello")
+            mocked_method.assert_called_once()
+
+
+class LoggerFrameTest(unittest.TestCase):
+    """Unit tests for the LoggerFrame class."""
+
+    def setUp(self):
+        self.qapp = QApplication([])
+
+    def tearDown(self):
+        del self.qapp
+
+    def test_button_ok_clicked(self):
+        with mock.patch('iquip.apps.logger.ConfirmClearingFrame.buttonOKClicked') as mocked_method:
+            app = logger.LoggerApp(name="name", parent=QObject())
+            app.confirmFrame.buttonBox.accepted.emit()
+            mocked_method.assert_called_once()
+
+    def test_button_cancel_clicked(self):
+        method_dir = 'iquip.apps.logger.ConfirmClearingFrame.buttonCancelClicked'
+        with mock.patch(method_dir) as mocked_method:
+            app = logger.LoggerApp(name="name", parent=QObject())
+            app.confirmFrame.buttonBox.rejected.emit()
+            mocked_method.assert_called_once()
+
+
+class LoggerAppTest(unittest.TestCase):
+    """Unit tests for the LoggerApp class."""
+
+    def setUp(self):
+        self.qapp = QApplication([])
+
+    def tearDown(self):
+        del self.qapp
+
+    def test_set_level(self):
+        """Tests for the LoggerApp's method setLevel.
+        
+        For vaild input, tests whether level of rootLogger and handler changes succesfully.
+        For invaild input, tests whether level of rootLogger and handler remains same.
+        """
+        levels = {"DEBUG": logging.DEBUG, "INFO": logging.INFO, "WARNING": logging.WARNING,
+                  "ERROR": logging.ERROR, "CRITICAL": logging.CRITICAL}
+        app = logger.LoggerApp('name')
+        root_logger = logging.getLogger()
+        for level, value in levels.items():
```

Comment:
Please rename them to `levelText` and `level`, the same as those in `logger.py`.